services/rest/websocket/connection/table_test.go

```diff
@@ -48,3 +48,66 @@ func TestStore(t *testing.T) {
 		assert.False(t, table.Exists(identification.Identifier{ID: "user1", Group: ""}))
 	})
 }
+
+func TestStoreEvent(t *testing.T) {
+	t.Run("Should store new event for a connection", func(t *testing.T) {
+		table := NewTable(10)
+		table.Store(identification.Identifier{ID: "user1", Group: ""})
+		table.StoreEvent(identification.Identifier{ID: "user1", Group: ""}, "event-id-1")
+
+		assert.True(t, table.HasEvent(identification.Identifier{ID: "user1", Group: ""}, "event-id-1"))
+	})
+
+	t.Run("Should not store new event if the connection is not active", func(t *testing.T) {
+		table := NewTable(10)
+		table.StoreEvent(identification.Identifier{ID: "user1", Group: ""}, "event-id-1")
+
+		assert.False(t, table.HasEvent(identification.Identifier{ID: "user1", Group: ""}, "event-id-1"))
+	})
+
+	t.Run("Should store multiple unique events for a connetion", func(t *testing.T) {
+		table := NewTable(10)
+		table.Store(identification.Identifier{ID: "user1", Group: ""})
+		table.StoreEvent(identification.Identifier{ID: "user1", Group: ""}, "event-id-1")
+		table.StoreEvent(identification.Identifier{ID: "user1", Group: ""}, "event-id-2")
+
+		assert.True(t, table.HasEvent(identification.Identifier{ID: "user1", Group: ""}, "event-id-1"))
+		assert.True(t, table.HasEvent(identification.Identifier{ID: "user1", Group: ""}, "event-id-2"))
+	})
+
+	t.Run("Should store multiple unique events for multiple connetion", func(t *testing.T) {
+		table := NewTable(10)
+		table.Store(identification.Identifier{ID: "user1", Group: ""})
+		table.Store(identification.Identifier{ID: "user2", Group: ""})
+
+		table.StoreEvent(identification.Identifier{ID: "user1", Group: ""}, "event-id-1")
+		table.StoreEvent(identification.Identifier{ID: "user1", Group: ""}, "event-id-2")
+		table.StoreEvent(identification.Identifier{ID: "user2", Group: ""}, "event-id-1")
+		table.StoreEvent(identification.Identifier{ID: "user2", Group: ""}, "event-id-2")
+
+		assert.True(t, table.HasEvent(identification.Identifier{ID: "user1", Group: ""}, "event-id-1"))
+		assert.True(t, table.HasEvent(identification.Identifier{ID: "user1", Group: ""}, "event-id-2"))
+		assert.True(t, table.HasEvent(identification.Identifier{ID: "user2", Group: ""}, "event-id-1"))
+		assert.True(t, table.HasEvent(identification.Identifier{ID: "user2", Group: ""}, "event-id-2"))
+	})
+
+	t.Run("Should remove all the events if connetion is removed or not active", func(t *testing.T) {
```

Comment:
Should we add a test to prove that once a connection is removed and added for the same user then the previous reqguids in the map are not there?

Reply:
Got it, as an extension to this test, we need additional asserts in line 112, where the same user is added and the the same "request-id-1" is added, assert that the event exists.

Reply:
done.